### PR TITLE
Warren lamb patch 1

### PR DESCRIPTION
**Pull request by Warren-Lamb**
actor entering  scene

### CODE REVIEW

File: Sources/Core/API/Scene.swift [hunk 1]
```diff
@@ -229,7 +229,9 @@ open class Scene: Pluggable, Activatable {
         let wasWithinScene = actor.isWithinScene
         let sceneRect = Rect(origin: .zero, size: size)
         actor.isWithinScene = actor.rect.intersects(sceneRect)
-
+        if !wasWithinScene && actor.isWithinScene {
```
Comment (JohnSundell) [inline Sources/Core/API/Scene.swift]: Can you surround this if statement with new lines to make the code a bit easier to read? 🙂

Comment (Warren-Lamb) [reply Sources/Core/API/Scene.swift]: You are a nice helpful person John.  By new lines do you just mean blank lines?  I will implement this using else if as you suggest.  I have no idea of how to implement a test in ActorTests so I will work on that now.  I used your movement tutorial and removed the boundary constraints and added an observer to the players new entered scene event during movement phase and logged it into the debug area.

Comment (JohnSundell) [reply Sources/Core/API/Scene.swift]: Looks great! Thanks a lot for this contribution! 🎉 And looks like this is your first time doing open source so hope you'll keep at it 😀

**Comment by Warren-Lamb** [comment]
Yipeee! That was fun and I learned some things about Github and testing. Thank you John for letting me contribute to your awesome game engine.  I will work on becoming a more proficient Swift programmer.  The comments in the test section for testEnteredScene in Actor tests should read 'top left' not center and below that 'mow' should read Now.

**Pull request merged**